spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -0,0 +1,280 @@
+    try {
+      return new SparkTable(icebergCatalog.createTable(
+          buildIdentifier(ident),
+          icebergSchema,
+          Spark3Util.toPartitionSpec(icebergSchema, transforms),
+          properties.get("location"),
+          properties));
+    } catch (AlreadyExistsException e) {
+      throw new TableAlreadyExistsException(ident);
+    }
+  }
+
+  @Override
+  public StagedTable stageCreate(Identifier ident, StructType schema, Transform[] transforms,
+                                 Map<String, String> properties) throws TableAlreadyExistsException {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    try {
+      return new StagedSparkTable(icebergCatalog.newCreateTableTransaction(buildIdentifier(ident), icebergSchema,
+          Spark3Util.toPartitionSpec(icebergSchema, transforms), properties.get("location"), properties));
+    } catch (AlreadyExistsException e) {
+      throw new TableAlreadyExistsException(ident);
+    }
+  }
+
+  @Override
+  public StagedTable stageReplace(Identifier ident, StructType schema, Transform[] transforms,
+                                  Map<String, String> properties) throws NoSuchTableException {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    try {
+      return new StagedSparkTable(icebergCatalog.newReplaceTableTransaction(buildIdentifier(ident), icebergSchema,
+          Spark3Util.toPartitionSpec(icebergSchema, transforms), properties.get("location"), properties,
+          false /* do not create */));
+    } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
+      throw new NoSuchTableException(ident);
+    }
+  }
+
+  @Override
+  public StagedTable stageCreateOrReplace(Identifier ident, StructType schema, Transform[] transforms,
+                                          Map<String, String> properties) {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    return new StagedSparkTable(icebergCatalog.newReplaceTableTransaction(buildIdentifier(ident), icebergSchema,
+        Spark3Util.toPartitionSpec(icebergSchema, transforms), properties.get("location"), properties,
+        true /* create or replace */));
+  }
+
+  @Override
+  public SparkTable alterTable(Identifier ident, TableChange... changes)
+      throws NoSuchTableException {
+    SetProperty setLocation = null;
+    SetProperty setSnapshotId = null;
+    SetProperty pickSnapshotId = null;
+    List<TableChange> propertyChanges = Lists.newArrayList();
+    List<TableChange> schemaChanges = Lists.newArrayList();
+
+    for (TableChange change : changes) {
+      if (change instanceof SetProperty) {
+        SetProperty set = (SetProperty) change;
+        if ("location".equalsIgnoreCase(set.property())) {
+          setLocation = set;
+        } else if ("current-snapshot-id".equalsIgnoreCase(set.property())) {
+          setSnapshotId = set;
+        } else if ("cherry-pick-snapshot-id".equalsIgnoreCase(set.property())) {
```

Comment:
Does it mean `ALTER TABLE t SET TBLPROPERTIES` can be used for rollbacks and cherry-picking? Will it make sense to build dedicated SQL commands for such operations and plugin via session extensions? I know it is much easier to do this via SET TBLPROPERTIES but I am not sure how natural it is.

Reply:
Yes, this is what we do to expose cherry-picking and setting the current snapshot explicitly through SQL. I can remove it if there are objections to this.

Reply:
These are very valuable features. Internally, we offer a SQL command to rollback tables. We don't expose cherry-picking yet but we are really excited about this feature. Do we want to have some SQL commands directly in Iceberg? I'd vote for that. It probably makes sense to agree on the syntax for common operations and use that across all query engines and all teams. For example, we can start with SNAPSHOT, MIGRATE, a command to rollback and a command to cherry-pick changes. These are easy to agree on. Later, we can build a VACUUM command that can optimize the table states. This one will be harder but we can add it later.

If that's something that we want to build, I can create a proposal in a couple of days.

Reply:
I'd love to see standardization here.